File: src/main/java/com/EmosewaPixel/pixellib/materialSystem/materials/MaterialStack.java
package com.EmosewaPixel.pixellib.materialSystem.materials;

//Material Stacks are ways of getting an amount of a certain Material
public class MaterialStack {
    private Material material;
    private int count;

    public MaterialStack(Material material, int count) {
        this.material = material;
        this.count = count;
    }

    public MaterialStack(Material material) {
        this(material, 1);
    }

    public Material getMaterial() {
        return material;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public boolean isEmpty() {
        return count == 0;
    }

    public MaterialStack copy() {
        return new MaterialStack(this.material, this.count);
    }
}

File: src/main/java/com/EmosewaPixel/pixellib/materialSystem/materials/MaterialStack.kt
package com.EmosewaPixel.pixellib.materialsystem.materials

//Material Stacks are ways of getting an amount of a certain Material
data class MaterialStack @JvmOverloads constructor(val material: Material, var count: Int = 1) {
    val isEmpty: Boolean
        get() = count == 0
}